jpegenc/subblocks/frontend/frontend_v2.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import numpy as np
+
+from itertools import chain
+import myhdl
+from myhdl import Signal, intbv, always_comb, always_seq, block, ResetSignal
+
+from jpegenc.subblocks.color_converters import ColorSpace, rgb2ycbcr_v2
+from jpegenc.subblocks.dct.dct_2d import dct_2d_transformation, dct_2d
+from jpegenc.subblocks.zig_zag import zig_zag_scan, zig_zag
+from jpegenc.subblocks.common import YCbCr_v2, input_interface, outputs_2d, RGB_v2, outputs_frontend_new
+
+
+def frontend_transform(blockr, blockg, blockb, N=8):
+    """software implementation of the frontend part"""
+    ycbcr_blocks = [[[] for _ in range(N)] for _ in range(3)]
+    dct_blocks, dct_blocks_linear, zig_zag_blocks = [[] for _ in range(3)]
+
+    """Color space conversion"""
```

Comment:
Using the multiline strings for comments `""" comment """` will have a negative impact on run-time (albeit small).  In python the `#` is used for comments and yes python doesn't have a multi-line comment escape.  The exception is the doc-strings, where the multiline string right after a function definition can be parsed by various tools (e.g. the python interrupter / interactive shell).

No need to change all the string comments now but something to be aware of (I should have commented on this in earlier PRs).